x/staketia/keeper/unbonding.go

```diff
@@ -61,11 +73,37 @@ func (k Keeper) RedeemStake(ctx sdk.Context, redeemer string, stTokenAmount sdkm
 	// Estimate a placeholder native amount with current RedemptionRate
 	// this estimate will be updated when the Undelegation record is finalized
 	nativeAmount := sdk.NewDecFromInt(stTokenAmount).Mul(stakeibcHostZone.RedemptionRate).TruncateInt()
+
+	// When checking if there's enough delegated TIA to handle the request,
+	// use the value from stakeibc instead of staketia
 	if nativeAmount.GT(stakeibcHostZone.TotalDelegations) {
 		return nativeToken, errorsmod.Wrapf(types.ErrUnbondAmountToLarge,
 			"cannot unstake an amount g.t. total staked balance: %v > %v", nativeAmount, stakeibcHostZone.TotalDelegations)
 	}
 
+	// Check if the requested unbonding is greater than what's in the multisig account
+	// If so, handle the spillover via stakeibc
+	if nativeAmount.GT(hostZone.RemainingDelegatedBalance) {
```

Comment:
It would be nice to be able to clean this up and instead just assume we need a last redemption that matches the amount perfectly.

Reply:
I don't have a strong opinion either way, but I think removing it leaves some messy edge cases and frontend complexity (e.g. what if a user tries to unbond a large amount). I think we discussed it with informal and they felt the same way

Here's the updated code [btw](https://github.com/Stride-Labs/stride/blob/07d8a838e664b4aa55859ab485f7a371ee6d7910/x/staketia/keeper/unbonding.go#L78)

I'd prefer to leave it as is cause I don't think removing it simplifies things _that_ much and we've already had it audited. But I could go either way

Reply:
yeah ig if it's already audited that's ok. don't think we'd hit the edge case because we could just submit the redemption ourselves when the account is low, but if it doesn't clean things up much maybe we just leave it.

Reply:
> don't think we'd hit the edge case because we could just submit the redemption ourselves when the account is low

I don't think we can just assume we'd be there to cover it. What if someone wants to unbond a large amount before the account gets low?

In that case, I think we'd have make changes to the frontend so that they unbond in two txs

Reply:
Wouldn't the worst case there be that their first tx fails, then they have to wait until we unbond the right amount (which I think would trigger the switch to the new module), then they could unbond their full amount?

Reply:
Depends how we want to implement it. Imo that'd be a risky approach. What if they want to unbond $1M - we can't handle that ourselves, so they'd be unable to unbond for a while

I think the way we'd want to implement it is to have the frontend notice that they're trying to redeem more than the remaining balance, and submit the first tx for just that amount, and then prompt them to unbond the remainder

Reply:
It's a good point about the large unbond. And the FE. I'm convinced. Resolving.